projects/arlas-components/src/lib/components/mapgl-import/mapgl-import.component.ts

```diff
@@ -613,14 +576,23 @@ export class MapglImportComponent {
   }
 
   public calcCentroid(feature) {
-    if (!this.maxVertexByPolygon) {
-      this.maxVertexByPolygon = 100;
-    }
-    if (this.maxVertexByPolygon && feature.geometry.coordinates[0].length - 1 > this.maxVertexByPolygon) {
-      this.tooManyVertex = true;
+    let cent;
+    if (feature.type === 'Point') {
+      cent = centroid(feature);
+    } else {
+      if (!this.maxVertexByPolygon) {
+        this.maxVertexByPolygon = 100;
+      }
+      if (this.maxVertexByPolygon && feature.geometry.coordinates[0].length - 1 > this.maxVertexByPolygon) {
+        this.tooManyVertex = true;
+      }
+
+      if (feature.type !== 'Point') {
```

Comment:
This condition is useless, here `feature.type` is necessarily different from `Point`, we are in the `else` of the opposite condition (L580)